src/Form/Type/AdminType.php

```diff
@@ -36,22 +37,22 @@
 class AdminType extends AbstractType
 {
     /**
-     * @var AdminHelper
+     * NEXT_MAJOR: Remove this property.
+     *
+     * @var AdminHelper|null
      */
     private $adminHelper;
 
     /**
-     * NEXT_MAJOR: Allow only `AdminHelper` for argument 1 and remove the default null value.
+     * NEXT_MAJOR: Remove the __construct method.
      */
     public function __construct(?AdminHelper $adminHelper = null)
     {
-        // NEXT_MAJOR: Remove this condition.
-        if (null === $adminHelper) {
+        if (\func_num_args() > 0) {
             @trigger_error(sprintf(
-                'Calling %s without passing an %s as argument is deprecated since sonata-project/admin-bundle 3.66'
+                'Passing argument 1 to %s() is deprecated since sonata-project/admin-bundle 3.x'
                 .' and will throw an exception in 4.0.',
```

Comment:
I guess passing a nonexistent argument won't throw an exception.
```suggestion
                'Passing argument 1 to %s() is deprecated since sonata-project/admin-bundle 3.x'
                .' and will be ignored in version 4.0.',
```